routes/matches/matches.js

```diff
@@ -525,24 +547,41 @@ router.get("/:match_id", async (req, res, next) => {
       req.user !== undefined &&
       (matchRow[0].user_id == req.user.id || Utils.superAdminCheck(req.user))
     ) {
-      sql = "SELECT * FROM `match` where id=?";
+//      sql = "SELECT * FROM `match` where id=?";
+	sql =
+        "SELECT `match`.id, `match`.user_id, server_id, team1_id, team2_id, winner, " +
+        "team1_score, team2_score, team1_series_score, team2_series_score, " +
+        "team1.name as team1_string, team2.name as team2_string, cancelled, forfeit, start_time, end_time, " +
+        "max_maps, title, skip_veto, private_match, enforce_teams, min_player_ready, " +
+        "season_id, is_pug, map_sides " +
+        "FROM `match`" +
+	"LEFT JOIN team team1 ON team1.id = `match`.team1_id " +
+        "LEFT JOIN team team2 ON team2.id = `match`.team2_id " +
+	"where `match`.id = ?";
     } else {
       sql =
-        "SELECT id, user_id, server_id, team1_id, team2_id, winner, " +
+	"SELECT `match`.id, `match`.user_id, server_id, team1_id, team2_id, winner, " +
         "team1_score, team2_score, team1_series_score, team2_series_score, " +
-        "team1_string, team2_string, cancelled, forfeit, start_time, end_time, " +
+        "team1.name as team1_string, team2.name as team2_string, cancelled, forfeit, start_time, end_time, " +
         "max_maps, title, skip_veto, private_match, enforce_teams, min_player_ready, " +
-        "season_id, is_pug, map_sides FROM `match` where id = ?";
+        "season_id, is_pug, map_sides " +
+        "FROM `match`" +
+        "LEFT JOIN team team1 ON team1.id = `match`.team1_id " +
+        "LEFT JOIN team team2 ON team2.id = `match`.team2_id " +
+        "where `match`.id = ?";
+
     }
 
     let matchID = req.params.match_id;
     let matches = await db.query(sql, matchID);
-
+    if (!matches.length) {
```

Comment:
Don't do this as connections will drop immediately instead of waiting for event changes.